apps/aecore/src/aec_sync.erl

```diff
@@ -158,6 +163,23 @@ handle_call({next_work_item, STId, PeerId, LastResult}, _From, State) ->
     {reply, Reply, State2};
 handle_call({gossip_txs, GossipTxs}, _From, State) ->
     {reply, ok, State#state{ gossip_txs = GossipTxs }};
+handle_call(sync_progress, _, State) ->
+    State#state.sync_tasks,
+    SyncProgress =
+        case State#state.sync_tasks of
+            [] -> 100.0;
```

Comment:
This means that the node will report 100% for a short while after starting, but I think that is ok.

Reply:
@hanssv Yes, this is rather approximate number... There are several issues:
* 100% when there are no peers in the config may be misleading, or just no sync tasks (not sure exactly when this could be the case, I haven't grasped all code paths around sync yet),
* relying on `aec_headers:height(aec_chain:top_header())` - we don't know if we're syncing with the chain that is based on the current top header in our DB?! On the other hand if we use height from `#sync_task.agreed` instead, that's not the height that has been already added to the database, but what will be added in the current sync worker step.... So that's not perfect as well... (And relying on `#sync_task.agreed` would save one db call for top header, shouldn't be an argument really 🙂 ).

I'd say this is good enough for now?...

Reply:
Why not reporting 0 on start, sounds much more reasonable to me?

Reply:
Ah, we can't differentiate between 100% and 0% .. I see.

Reply:
Yes, technically, we just don't have sync task (maybe yet... maybe already...). We could try to code some bigger machinery, but I'm not sure if that's worth doing.